### DISCUSSION TITLE
Control Edge API Precedence?

### DISCUSSION THREAD

**Discussion by carlreid** [vercel/next.js#40255] [Help]
If I have a set up such as:
```js
/pages/api/[...gateway].ts // Using Edge API, forwards requests to another API
/pages/api/health.ts // Using normal API route, returns a health check for the frontend's backend
```

Currently the `[...gateway].ts` file seems to be handling the `/api/health` route, which seems to break the [precedence rules](https://nextjs.org/docs/api-routes/dynamic-api-routes#caveats)?   
  
Do I need to configure something for this? Doesn't seem like anything on the [PageConfig](https://github.com/vercel/next.js/blob/0796b6faa991cd58a517b0f160320e9978208066/packages/next/types/index.d.ts#L61-L91) matches this need.

**Comment by icyJoseph** [comment]
Could you provide a reproduction demo? Sure sounds like a bug... 

## Edit

Fixed on `12.2.6-canary.10`, but it is present on `12.2.5`. Perhaps it was fixed on an earlier canary, but it is best to way for the next full release.

**Reply by carlreid** [reply]
I hadn't created one, but this is causing the same issue here: https://github.com/carlreid/next-routing-example

It's just:
1) `yarn create next-app --typescript`
2) Navigate to `http://localhost:3000/api/hello`
3) Great, it says "John Doe"
4) Create `/pages/api/[...gateway].ts` with example from the docs:  
```ts
export const config = {
  runtime: "experimental-edge",
};

export default (req) => new Response('Hello world!')
```
5) Refresh/navigate to `http://localhost:3000/api/hello`
6) Now it says "Hello world!", we'd expect "John Doe" still.

**Reply by icyJoseph** [reply]
Alright, yeah looks like a bug in the latest release, but it has been fixed on `next@canary`, 12.2.6-canary.10 to be precise. so we just have to wait for the next release.

**Reply by carlreid** [reply]
Actually looks like it was fixed in 12.2.6-canary.1, this PR: https://github.com/vercel/next.js/pull/39594
> Another fix is to make sure Edge Function routes are sorted too, so /foo can be caught before dynamic ones like /[id].

Well,  nice to know I am not just doing something stupid then 😁
